Revise densities & atomic weights for chemical elements to match Geant4

cvuosalo opened the pull request:
`materials.xml` has contained for many years some non-standard or nonsense values for the densities and atomic weights of certain chemical elements. To avoid any confusion, we are standardizing these values to match those used by Geant4. References for the Geant4 values are as follows.

Atomic weights from NIST: https://www.nist.gov/pml/periodic-table-elements

Densities: http://geant4-userdoc.web.cern.ch/geant4-userdoc/UsersGuides/ForApplicationDeveloper/html/Appendix/materialNames.html

2021 DDD and DD4hep scenarios are updated to use the new version of `materials.xml`. Going forward, all new Run 3 and Phase 2 scenarios should use this new version.

#### PR validation:

Test DB payloads were created with the new materials file and compared with existing DB payloads to check for errors.

Workflow 11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021 was run using XML geometry with and without this PR. The DQM results show no significant differences.

No backport is needed.

cvuosalo committed:
Revise vales for chemical elements to match Geant4

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29856/15408

- This PR adds an extra 36KB to repository

cmsbuild commented:
A new Pull Request was created by  @cvuosalo (Carl Vuosalo) for master.

It involves the following packages:

Geometry/CMSCommonData


@civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @kpedro88 can you please review it and eventually sign? Thanks.
@fabiocos, @slomeo this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: bf7e68b6d1021bbf3e6d1db8782525cffdb06297
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-85b897/6347/summary.html
CMSSW: CMSSW_11_1_X_2020-05-15-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-85b897/6347/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2702162
 - DQMHistoTests: Total failures: 12
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2701831
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 150 log files, 16 edm output root files, 35 DQM output files

civanch commented:
@cvuosalo , I would agree with fixes in many of materials, however, there are questions:

1) Carbon is the most difficult material, because it may be in different forms from graphite to brilliant with different densities. What is used in CMS is a problem, which need to be evaluated.
2) Bor10 and Bor11 are defined as a separate materials, what is used in CMS?
3) Deuterium - is it used anywhere?

fabiocos commented:
@cvuosalo @civanch thanks for attacking this issue. I believe that an important practical check is to verify the impact of any change on our material budget evaluation. Was this already tested?

cvuosalo commented:
@fabiocos  No, there has been no material budget test. Who could perform this test? I would need instructions.

davidlange6 commented:
its not obvious to me why standard densities should be always used in the simulation. (or are these densities not actually used by g4?)

cvuosalo commented:
@civanch 
1. I'm not sure if the elemental Carbon density is ever used. Composite materials seem to specify their own densities. Various graphite materials also specify their own densities.
2. Bor 10 and Bor 11 are used in materials like "AQBB borated concre", "Heavy Boron concrete", and "B_2 O_3". Heavy boron concrete is used in the muon yoke, and Bor 11 is used in the "TEC_PitchAdapter" in the Tracker.
3. Deuterium is used as a target in `​SimG4Core/​CheckSecondary/​data/​DTarget.xml`.

cvuosalo commented:
@davidlange6 I think the density of elements that exist in pure form in the detector are used (e.g. copper). But if an element only occurs in compounds, then the elemental density is probably never used.

civanch commented:
@cvuosalo , because there is no change of results I suspect, that all what is changed is not used in baseline simulation. 

These material descriptions are historical. Even if many of that materials are not used in CMS directly better to keep new values coherent with Geant4.

civanch commented:
These updated materials are in v1 sub-directory, so do not break anything.

civanch commented:
+1

cvuosalo commented:
urgent

cvuosalo commented:
We would like this PR to go into 11_1_pre8.

davidlange6 commented:
I guess what you mean is that you are updating only the xml geometry and not the GT one so that there is no effect. That doesn’t seem like a good approach as it leaves surprises for later.

On May 19, 2020, at 5:35 PM, Vladimir Ivantchenko <notifications@github.com<mailto:notifications@github.com>> wrote:



These updated materials are in v1 sub-directory, so do not break anything.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/29856#issuecomment-630901992>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQ7PISBF4SPWREP5THDRSKRLHANCNFSM4NCSNUKA>.

civanch commented:
@davidlange6 , it may be used or not for run-3 - old material.xml is not removed and we need to discuss what is optimal. This new file may be useful for Phase-2, why use very old values for Phase-2? At the same time, the modification is really small, there is no one big fix, so no significant change of results are expected.

cvuosalo commented:
@davidlange6 This PR should be the last step of revising Run 3 geometry. There have been several geometry revisions in the last few weeks. We decided to bundle them into a single Global Tag update. With the merging of this PR and the closing of 11_1 to new development, I will create new 2021 geometry DB payloads and a new Global Tag.

I have already done preliminary tests that show that there are no major differences between 2021 DB and 11_1 XML geometry.

davidlange6 commented:
Its instead a question of showing that the new values are better.  Should these not get added to the run 3 GT (as thats the geometry that got updated) to see their effect?

On May 19, 2020, at 5:54 PM, Vladimir Ivantchenko <notifications@github.com<mailto:notifications@github.com>> wrote:



@davidlange6<https://github.com/davidlange6> , it may be used or not for run-3 - old material.xml is not removed and we need to discuss what is optimal. This new file may be useful for Phase-2, why use very old values for Phase-2? At the same time, the modification is really small, there is no one big fix, so no significant change of results are expected.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/29856#issuecomment-630912887>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQ5MCV7BALGME6BDZA3RSKTTJANCNFSM4NCSNUKA>.

cvuosalo commented:
@davidlange6 I did test this PR checking the DQM and physics values from workflow 11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021 using geometry XML and found no significant differences. After I create new geometry DB payloads for 2021, they will be tested, but those will be overall tests of all geometry changes since the previous geometry DB payloads were created, and not just this PR.

davidlange6 commented:
Thanks - this approach makes sense.. Sounds like the PR for the new GT will come soon.

On May 19, 2020, at 6:59 PM, Carl Vuosalo <notifications@github.com<mailto:notifications@github.com>> wrote:



@davidlange6<https://github.com/davidlange6> I did test this PR checking the DQM and physics values from workflow 11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021 using geometry XML and found no significant differences. After I create new geometry DB payloads for 2021, they will be tested, but those will be overall tests of all geometry changes since the previous geometry DB payloads were created, and not just this PR.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/29856#issuecomment-630950857>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQ6KLRYGT55XZTDYGPTRSK3HXANCNFSM4NCSNUKA>.

kpedro88 commented:
+upgrade

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
